source/content/guides/secrets/04-use-cases.md

```diff
@@ -0,0 +1,225 @@
+---
+title: Pantheon Secrets Guide
+subtitle: Use Cases
+description: Some common uses cases for Pantheon Secrets
+terminuspage: true
+contributors: [stovak]
+contenttype: [guide]
+innav: [true]
+categories: [secrets]
+cms: [drupal, wordpress]
+audience: [development]
+product: [secrets]
+integration: [--]
+tags: [reference, cli, local, terminus, workflow]
+permalink: docs/guides/secrets/use-cases
+reviewed: "2024-05-01"
+---
+
+# Use Case: Using secrets with Integrated Composer
+
+## Mechanism 1: Oauth composer authentication (recommended)
+
+#### GitHub Repository
+
+1. [Generate a Github token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token). The Github token must have all "repo" permissions selected.
+
+    NOTE: Check the repo box that selects all child boxes. **Do not** check all child boxes individually as this does not set the correct permissions.
+
+    ![image](https://user-images.githubusercontent.com/87093053/191616923-67732035-08aa-41c3-9a69-4d954ca02560.png) 
+
+1. Set the secret value to the token via terminus: `terminus secret:site:set <site> github-oauth.github.com <github_token> --type=composer --scope=user,ic`
+
+1. Add your private repository to the `repositories` section of `composer.json`:
+
+    ```json
+    {
+        "type": "vcs",
+        "url": "https://github.com/your-organization/your-repository-name"
+    }
+    ```
+
+    Your repository should contain a `composer.json` that declares a package name in its `name` field. If it is a WordPress plugin or a Drupal module, it should specify a `type` of `wordpress-plugin` or `drupal-module` respectively. For these instructions, we will assume your package name is `your-organization/your-package-name`.
+
+1. Require the package defined by your private repository's `composer.json` by either adding a new record to the `require` section of the site's `composer.json` or with a `composer require` command:
+
+    ```bash
+    composer require your-organization/your-package-name
+    ```
+
+1. Commit your changes and push to Pantheon.
+
+#### GitLab Repository
```

Comment:
Maybe Gitlab and Bitbucket examples can become tabs, along with the previous "Github" section. Otherwise, the content below is too repetitive.

Reply:
I tried to get them to work, but they don't seem to be working right. I'll get Rachel to correct my problem, but the tabs are there delineated.

Reply:
https://github.com/pantheon-systems/documentation/pull/8975/commits/940130004448dcdf995cad6633822238b93986a9